pilot/pkg/xds/eds.go

```diff
@@ -456,15 +456,21 @@ func shouldUseDeltaEds(req *model.PushRequest) bool {
 // onlyEndpointsChanged checks if a request contains *only* endpoints updates. This allows us to perform more efficient pushes
 // where we only update the endpoints that did change.
 func onlyEndpointsChanged(req *model.PushRequest) bool {
-	if len(req.ConfigsUpdated) > 0 {
-		for k := range req.ConfigsUpdated {
-			if k.Kind != kind.ServiceEntry {
-				return false
-			}
+	// If we don't know what configs are updated, just send a full push
+	if len(req.ConfigsUpdated) == 0 {
+		return false
+	}
+	for cfg := range req.ConfigsUpdated {
+		if _, f := skippedEdsConfigs[cfg.Kind]; f {
```

Comment:
I can see we have already checked this in `edsNeedsPush` which is called before `onlyEndpointsChanged`. So why do we need check again?

Reply:
Let's say a service and a VS are updated in one push request
|   |  edsNeedsPush | onlyEndpointsChanged  |
| :------------: | :------------: | :------------: |
| Before  |  true | false  |
|  After | true  |  true |

Reply:
Get it, but this violates the function name.

Reply:
Not sure if used in other place this is till right